clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -53,7 +53,7 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals, boolean doLog)
                 throw new ConfigException(key.toString(), originals.get(key), "Key must be a string.");
         this.originals = (Map<String, ?>) originals;
         this.values = definition.parse(this.originals);
-        this.used = Collections.synchronizedSet(new HashSet<String>());
+        this.used = Collections.synchronizedSet(new HashSet<>());
```

Comment:
Yeah, inference doesn't work in that case. Very lame! Are you saying that they improved that in Java 8?